manual/embedded-examples/src/docs/dev/procedures.asciidoc

```diff
@@ -5,69 +5,46 @@
+User-defined procedures are written in Java, deployed into the database, and called from Cypher.
 --
 
-A procedure is a means of extending Neo4j by writing custom code which can be invoked directly from Cypher.
-It can take arguments, perform operations on the database, and return results.
-The procedure is written in the Java programming language and compiled into a _jar_ file.
-A procedure is deployed to the database by dropping the _jar_ file into the _$NEO4J_HOME/plugins_ directory on each server in the cluster.
-The databases must be re-started to pick up the procedure.
+A _procedure_ is a mechanism that allows Neo4j to be extended by writing custom code which can be invoked directly from Cypher.
+Procedures can take arguments, perform operations on the database, and return results.
+
+Procedures are written in Java and compiled into _jar_ files.
+They can be deployed to the database by dropping a _jar_ file into the _$NEO4J_HOME/plugins_ directory on each standalone or clustered server.
+The database must be re-started on each server to pick up new procedures.
 
 Procedures are the preferred means for extending Neo4j.
 Examples of use cases for procedures are:
 
-. To provide access to a product surface that has not yet been opened to Cypher.
-  This includes manual indexes and schema introspection.
-. To perform graph-global operations, for example counting connected components or finding dense nodes.
+. To provide access to functionality that is not available in Cypher, such as manual indexes and schema introspection.
+. To provide access to third party systems.
+. To perform graph-global operations, such as counting connected components or finding dense nodes.
 . To express a procedural operation that is difficult to express declaratively with Cypher.
 
 
-== Using a procedure
-
-To call a stored procedure, use the Cypher `CALL` clause.
-A procedure named `findDenseNodes` defined in the package `org.neo4j.examples` is called using `CALL org.neo4j.examples.findDenseNodes(1000)`.
-See the Cypher documentation on <<query-call, the `CALL` clause.>>
+== Calling procedures
 
-[NOTE]
---
-A query with a `CALL` clause cannot include any other clauses.
---
+To call a stored procedure, use a Cypher `CALL` clause.
+The procedure name must be fully qualified, so a procedure named `findDenseNodes` defined in the package `org.neo4j.examples` could be called using:
 
-The `CALL` example illustrates a reasonable use case for a procedure: to locate _dense nodes_.
-Dense nodes are nodes that have a large number of incoming or outgoing relationships.
-Finding dense nodes is a graph-global operation and a good fit for procedural code, to be called from Cypher.
-
-Continuing with the `CALL` example, a procedure to locate dense nodes might take a `threshold` argument, defining how many relationships are necessary for a node to be considered dense.
-
-.Find dense nodes procedure
-[snippet, java]
+[snippet, cypher]
```

Comment:
This should be `[source, cypher]`, since the block contains inline source code. The `snippet` block style is used when including source code snippets from some other file, and the block contains not inline source code but the coordinates for the source code to include.